### DISCUSSION TITLE
How to Disable a Tenant

### DISCUSSION THREAD

**Discussion by iamunadike** [django-tenants/django-tenants#787] [Ideas]
@tomturner ,
I wish to ask if presently it is possible to disable a tenant-
such that every user with the tenant does not have access if set to a say inactive or disabled
such a feature will be very great for dajngo-tenants, since it leverages more control per schema

**Comment by tomturner** [comment]
It's quite easy todo already.  Infact I do it my app.   You create the field on the domain and then create you own custom tenant middware which checks that flag.  In my case I have something like this in my middleware

    if not has_error and not self.is_tenant_enabled(request.tenant):
            request.urlconf = settings.DISABLED_URLCONF
            if not request.path == '/':
                return HttpResponseRedirect('/')

**Reply by iamunadike** [reply]
I think we should add this to the middleware too and the is_tenant_disabled field option so it is easy
in this case we dont have to return a redirect, we just raise a validation error, that error, can we read and intepreted into anythin including a redirect

**Reply by tomturner** [reply]
To be honest I can't see how anyone can use DT without a custom middleware as everyone wants somthing differnent

**Reply by iamunadike** [reply]
despite, it is still valid, to have a generic way to lock the tenant is disabled and then reactivate when enabled
